### ISSUE TITLE
Titanic case study using logistic case study

### ISSUE THREAD

**Issue by Iamsachinpatil** [Iamsachinpatil/machine-learning#1]
`import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

import os
os.chdir('D:/Jigsaw Academy/machine learning')

Train=pd.read_csv('train.csv',sep=',',header=0,encoding='latin')
Test=pd.read_csv('test.csv',sep=",",header=0,encoding='latin')

titnic=pd.concat([Train,Test],sort=False)
titnic.head()

print(titnic.shape)
titnic.describe()

titnic.info()

titnic.isnull().sum()

titnic['Age']=titnic['Age'].fillna(np.mean(titnic['Age']))
titnic['Fare']=titnic['Fare'].fillna(np.mean(titnic['Fare']))

titnic['Embarked'] = titnic['Embarked'].fillna('S')

titnic1 = titnic.drop(['Name', 'Ticket', 'Cabin'], axis = 1)
titnic1.isnull().sum()

titnic2 = pd.get_dummies(titnic1, drop_first = True)
titnic2.head(1)

t2_train = titnic2[titnic2['Survived'].notna()]
t2_train.info()

t2_test=titnic2[titnic2['Survived'].isna()]
t2_test.info()

X = t2_train.drop(['PassengerId','Survived'],axis=1)

y = t2_train['Survived']

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)

from sklearn.linear_model import LogisticRegression
clf = LogisticRegression(random_state=0,solver='liblinear').fit(X, y)

clf.predict_proba(X)

clf.score(X, y)

predictions = clf.predict(X_test)
predictions

from sklearn.metrics import confusion_matrix
confusion_matrix(y_test, predictions)

from sklearn.metrics import classification_report
print(classification_report(y_test,predictions))`

**Comment by Iamsachinpatil** [comment]
> `import numpy as np
> import pandas as pd
> import seaborn as sns
> import matplotlib.pyplot as plt
> %matplotlib inline
> 
> import os
> os.chdir('D:/Jigsaw Academy/machine learning')
> 
> Train=pd.read_csv('train.csv',sep=',',header=0,encoding='latin')
> Test=pd.read_csv('test.csv',sep=",",header=0,encoding='latin')
> 
> titnic=pd.concat([Train,Test],sort=False)
> titnic.head()
> 
> print(titnic.shape)
> titnic.describe()
> 
> titnic.info()
> 
> titnic.isnull().sum()
> 
> titnic['Age']=titnic['Age'].fillna(np.mean(titnic['Age']))
> titnic['Fare']=titnic['Fare'].fillna(np.mean(titnic['Fare']))
> 
> titnic['Embarked'] = titnic['Embarked'].fillna('S')
> 
> titnic1 = titnic.drop(['Name', 'Ticket', 'Cabin'], axis = 1)
> titnic1.isnull().sum()
> 
> titnic2 = pd.get_dummies(titnic1, drop_first = True)
> titnic2.head(1)
> 
> t2_train = titnic2[titnic2['Survived'].notna()]
> t2_train.info()
> 
> t2_test=titnic2[titnic2['Survived'].isna()]
> t2_test.info()
> 
> X = t2_train.drop(['PassengerId','Survived'],axis=1)
> 
> y = t2_train['Survived']
> 
> from sklearn.model_selection import train_test_split
> X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)
> 
> from sklearn.linear_model import LogisticRegression
> clf = LogisticRegression(random_state=0,solver='liblinear').fit(X, y)
> 
> clf.predict_proba(X)
> 
> clf.score(X, y)
> 
> predictions = clf.predict(X_test)
> predictions
> 
> from sklearn.metrics import confusion_matrix
> confusion_matrix(y_test, predictions)
> 
> from sklearn.metrics import classification_report
> print(classification_report(y_test,predictions))`
